### ISSUE TITLE
Albert model? I will be so pleased to test the ALBert model in tf2. 

### ISSUE THREAD

**Issue by YankeeMarco** [tensorflow/models#7741]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

Also, please understand that many of the models included in this repository are experimental and research-style code. If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **What is the top-level directory of the model you are using**:
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

`python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by saberkun** [comment]
Hi, Albert is straightforward to implement. We are releasing a newer version soon including Albert variants. Assigning the issue to me. Will update here.
Thanks!

**Comment by saberkun** [comment]
We are working in progress. Expect to release checkpoints and models soon.

**Comment by stefan-it** [comment]
Hi guys,

is there any news about an ALBERT pretraining with Models garden?

Many thanks!

**Comment by laxmareddyp** [comment]
Hi @stefan-it ,

Could you please check this [Albert pretrained_models.md](https://github.com/tensorflow/models/blob/master/official/nlp/docs/pretrained_models.md)  from the Model Garden will help you.

**Comment by stefan-it** [comment]
Hi @laxmareddyp,

thanks for your quick reply! So my aim is to pretrain an ALBERT model from scratch. I could sucessfully pretrain BERT, BERT with Token Dropping and TEAMS models with this great Model Garden library here. E.g. this is pretraining command for BERT with Token Dropping:

```bash
python3 train.py \
    --experiment=token_drop_bert/pretraining \
    --config_file=gwlms_base_pretrain_sequence_pack.yaml \
    --config_file=gwlms_base_token_drop.yaml \
    --params_override="runtime.distribution_strategy=tpu" \
    --tpu=gwlms \
    --model_dir=gs://gwlms/models/bert-base-td-german-wikipedia-v1-dupe5-cased \
    --mode=train
```

As far as I can see, there's no experiment definition like `albert/pretraining`. Technically, the generation of pretraining data should work for ALBERT, as the `create_pretraining_data.py` [script](https://github.com/tensorflow/models/blob/master/official/nlp/data/create_pretraining_data.py#L42) supports SPM models, so I think "only" the experiment definition part for ALBERT is missing here :thinking:

**Comment by stefan-it** [comment]
Hi @laxmareddyp ,

I tried pretraining of ALBERT now, with this configuration:

```yaml
task:
  init_checkpoint: ''
  model:
    cls_heads: [{activation: tanh, cls_token_idx: 0, dropout_rate: 0.1, inner_dim: 768, name: next_sentence, num_classes: 2}]
    encoder:
      type: albert
      albert:
        attention_dropout_rate: 0.0
        dropout_rate: 0.0
        hidden_activation: gelu
        hidden_size: 768
        initializer_range: 0.02
        intermediate_size: 3072
        max_position_embeddings: 512
        num_attention_heads: 12
        num_layers: 12
        type_vocab_size: 2
        vocab_size: 30000
  train_data:
    drop_remainder: true
    global_batch_size: 512
    input_path: 'gs://stefan-experiments/tfrecords-gwlms/*.tfrecord'
    is_training: true
    max_predictions_per_seq: 76
    seq_length: 512
    use_next_sentence_label: true
    use_position_id: false
    use_v2_feature_names: true
trainer:
  checkpoint_interval: 10000
  max_to_keep: 3000
  optimizer_config:
    learning_rate:
      polynomial:
        cycle: false
        decay_steps: 125000
        end_learning_rate: 0.0
        initial_learning_rate: 0.00176
      type: polynomial
    optimizer:
      type: lamb
    warmup:
      linear:
        warmup_steps: 3125
      type: linear
  steps_per_loop: 1000
  summary_interval: 1000
  train_steps: 125000
  validation_interval: 1000
  validation_steps: 64
```

I thought that I could just use the pretraining command that is similar for pretraining a BERT model:

```bash
python3 train.py --experiment=bert/pretraining \
--config_file=gwlms_pretrain.yaml \
--params_override=runtime.distribution_strategy=tpu \
--tpu=albert \
--model_dir=gs://stefan-experiments/models/gwlms-albert-base \
--mode=train
```

I created the pretraining data with an ALBERT tokenizer model (so SPM instead of WordPiece), I thought it would be working (the only fundamental change w.r.t. to the original ALBERT training is, that I use NSP instead of SPO, which is ok for my use case now).

However, the pretraining crashed, because of loss NaN, and it saw the following warning:

```
/tmp/__autograph_generated_filearhpbz2f.py:63: UserWarning: Input dict contained keys ['masked_lm_positions', 'masked_lm_ids', 'masked_lm_weights', 'next_sentence_labels'] which did not match any model input. They will be ignored by the model.
```

Do you accidentally know how to get ALBERT pretraining running? I do not necessarily need the SOP training objective, NSP would be fine :)

Many thanks in advance!

**Comment by stefan-it** [comment]
Quick update on that: pretraining is possible, I made only two mistakes: a) vocab size was not correct, as I've trained an own SPM model with 32k - not 30k and b) `cls_token_idx: 2` -> the correct index of the `[CLS]` token needs to be set, standard BERT wordpiece model has `[CLS]` at position 0.

I could sucessfully pretrain an ALBERT base model, I've also written a conversion script TF Model Garden -> HF Transformers and the MLM predictions look very good.
